webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,288 @@
+        if (num > str.length) {
+            return str;
+        }
+        return `${str.substring(0, num)}...`;
+    }
+
+    constructor(Props: Props) {
+        super(Props);
+        let ticketID = '';
+        if (this.props.href.includes('selectedIssue')) {
+            ticketID = this.props.href.split('selectedIssue=')[1].split('&')[0];
+        }
+
+        if (!ticketID && this.props.href.includes('atlassian.net/browse')) {
+            ticketID = this.props.href.split('|')[0].split('?')[0].split('/browse/')[1];
+        }
+
+        this.state = {
+            href: this.props.href,
+            isLoaded: false,
+            ticketId: ticketID,
+        };
+    }
+
+    init() {
+        let instanceID = '';
+        if (this.props.connectedInstances.length === 1) {
+            instanceID = this.props.connectedInstances[0].instance_id;
+        } else if (this.props.defaultUserInstanceID) {
+            instanceID = this.props.defaultUserInstanceID;
+        }
+
+        let ticketID = '';
+        if (this.props.href.includes('selectedIssue')) {
+            ticketID = this.props.href.split('selectedIssue=')[1].split('&')[0];
+            this.props.getIssueByKey(ticketID, instanceID);
+        }
+
+        if (!ticketID && this.props.href.includes('atlassian.net/browse')) {
+            ticketID = this.props.href.split('|')[0].split('?')[0].split('/browse/')[1];
+            if (ticketID) {
+                this.props.getIssueByKey(ticketID, instanceID);
+            }
+        }
+    }
+
+    componentDidMount() {
+        if (this.props.connected && !this.state.isLoaded && ((this.props.ticketDetails && this.props.ticketDetails.ticketId) !== this.state.ticketId || !this.props.isLoaded)) {
+            this.init();
+        } else if (this.props.connected && !this.state.isLoaded && this.props.ticketDetails && this.props.ticketDetails.ticketId === this.state.ticketId) {
+            this.setTicket(this.props);
+        }
+    }
+
+    componentDidUpdate() {
+        if (this.props.isLoaded && !this.state.isLoaded && this.props.ticketDetails && this.props.ticketDetails.ticketId === this.state.ticketId) {
+            this.setTicket(this.props);
+        }
+    }
+
+    setTicket(data: Props) {
+        this.setState({
+            isLoaded: true,
+            ticketDetails: data.ticketDetails,
+        });
+    }
+
+    fixVersionLabel(fixVersion: string) {
+        if (fixVersion) {
+            const fixVersionString = 'Fix Version :';
+            return (
+                <div className='fix-version-label'>
+                    {fixVersionString}
+                    <span className='fix-version-label-value'>
+                        {fixVersion}
+                    </span>
+                </div>
+            );
+        }
+
+        return (<span/>);
+    }
+
+    tagTicketStatus(ticketStatus: string) {
+        let ticketStatusClass = 'default-style';
+
+        switch (ticketStatus.toLowerCase()) {
```

Comment:
I'd prefer this approach, less code, easier to maintain and update:

```typescript
const myStatusClasses = {
  statusIndeterminate: 'ticket-status--indeterminate',
  statusDone: 'ticket-status-done',
  ....

 Then you can use it like:

try {
 myStatusClasses[ticketStatus.toLowerCase]
} catch { ... // default }
```

Using switch cases is almost always a bad idea in these cases, it's much easier to maintain a data struct and self-documented, easier to read and much less code to maintain (imagine having 20 statuses, for example).